### PR TITLE
Update

### PR DESCRIPTION
**Pull request by jasongrout**
This updates to TS 2.0 and cleans up a lot of the typings, etc.

### CODE REVIEW

**Comment by jasongrout** [comment]
@blink1073 - I think this is ready for review.

File: test/src/typings.d.ts [hunk 1]
```diff
@@ -1,5 +0,0 @@
-/// <reference path="../../typings/expect.js/expect.js.d.ts"/>
-/// <reference path="../../typings/es6-promise/es6-promise.d.ts"/>
-/// <reference path="../../typings/mocha/mocha.d.ts"/>
-/// <reference path="../../typings/node/node.d.ts"/>
-/// <reference path="../../typings/webpack/webpack.d.ts"/>
```
Comment (blink1073) [inline test/src/typings.d.ts:5]: Why don't we just augment the webpack typings locally for now?

Comment (jasongrout) [reply test/src/typings.d.ts:5]: Good idea, if I can get into that namespace inside the typings. I'll try.

Comment (jasongrout) [reply test/src/typings.d.ts:5]: Done. Thanks, that cleaned things up nicely.

**Comment by blink1073** [comment]
This is an impressive add/delete ratio, thanks!

**Pull request merged**